src/centered-on-asteroid/model_def.py

```diff
@@ -1,6 +1,69 @@
 import torch
 import torch.nn as nn
-from typing import Tuple
+from typing import Tuple, List
+
+
+class DynamicCNN(nn.Module):
+    def __init__(
+        self,
+        image_shape: Tuple[int, int] = (30, 30),
+        num_conv_blocks: int = 2,
+        filters_list: List[int] = [16, 32],
+        kernel_size: int = 3,
+        stride: int = 1,
+        padding: int = 1,
+        feature_vector_output_size: int = 1,
+    ):
+        super(DynamicCNN, self).__init__()
+        self.image_shape = image_shape
+
+        self.num_conv_blocks = num_conv_blocks
+        self.filters_list = filters_list
+        self.kernel_size = kernel_size
+        self.stride = stride
+        self.padding = padding
+
+        self.maxpool_kernel_size = 2
+        self.maxpool_stride = 2
+
+        self.conv_blocks = nn.ModuleList(
+            [nn.Conv2d(1, filters_list[0], kernel_size, stride, padding)]
```

Comment:
If you require the filters_list to start with 1, you don't need to handle this case separately